addons/payment/static/src/js/payment_form.js

```diff
@@ -0,0 +1,250 @@
+                            else {
+                                // we just go to the return_url or reload the page
+                                if(form_data.return_url) {
+                                    window.location = form_data.return_url;
+                                }
+                                else {
+                                    window.location.reload();
+                                }
+                            }
+                        }
+                        // if the server has returned false, we display an error
+                        else {
+                            self.error(core._t('Error'),
+                                core._t("<p>We are not able to add your payment method at the moment.<br/> Please try again later or contact us.</p>"));
+                        }
+                        // here we remove the 'processing' icon from the 'add a new payment' button
+                        $(button).attr('disabled', false);
+                        $(button).find('i:first').remove();
+                    }).fail(function(message, data) {
+                        // if the rpc fails, pretty obvious
+                        $(button).attr('disabled', false);
+                        $(button).find('i:first').remove();
+
+                        self.error(core._t('Error'),
+                            core._t("<p>We are not able to add your payment method at the moment.<br/> Please try again later or contact us.</p>") + (core.debug ? data.data.message : ''));
+                    });
+                }
+                else {
+                    this.error(core._t('Error'),
+                        core._t('<p>Please select the option to add a new payment method</p>'));
+                }
+            },
+            // event handler when clicking on a button to delete a payment method
+            'click button[name="delete_pm"]': function (ev) {
+                ev.stopPropagation();
+                ev.preventDefault();
+                var self = this;
+                var pm_id = ev.target.value;
+
+                var Delete = function() {
+                    ajax.jsonRpc('/payment/delete/', 'call', {'pm_id':pm_id}).then(function(data) {
+                        if(data === true) {
+                            ev.target.closest('div').remove();
+                        }
+                        else {
+                            self.error(core._t('Error'),
+                                core._t("<p>We are not able to delete your payment method at the moment.<br/> Please try again later or contact us.</p>"));
+                        }
+                    }).fail(function(message, data){
+                        self.error(core._t('Error'),
+                            core._t("<p>We are not able to add your payment method at the moment.<br/> Please try again later or contact us.</p>") + (core.debug ? data.data.message : ''));
```

Comment:
also you can remove the 'try again later' 
personnally i would found it irritating if a form told me to try again later if no action whatsoever is taken to actually resolve the issue :p